src/js_sync.hpp

```diff
@@ -724,26 +721,35 @@ void SyncClass<T>::set_sync_log_level(ContextType ctx, ObjectType this_object, A
 
     auto app = *get_internal<T, AppClass<T>>(ctx, Value::validated_to_object(ctx, args[0], "app"));
     std::string log_level = Value::validated_to_string(ctx, args[1], "log level");
-    std::istringstream in(log_level); // Throws
-    in.imbue(std::locale::classic()); // Throws
-    in.unsetf(std::ios_base::skipws);
-    util::Logger::Level log_level_2 = util::Logger::Level();
-    in >> log_level_2; // Throws
-    if (!in || !in.eof())
-        throw std::runtime_error("Bad log level");
-    app->sync_manager()->set_log_level(log_level_2);
+
+    auto level = common::logger::Logger::get_level(log_level);
+    app->sync_manager()->set_log_level(level);
 }
 
-#if REALM_PLATFORM_NODE
 template<typename T>
 void SyncClass<T>::set_sync_logger(ContextType ctx, ObjectType this_object, Arguments &args, ReturnValue &return_value) {
     args.validate_count(2);
 
     auto app = *get_internal<T, AppClass<T>>(ctx, Value::validated_to_object(ctx, args[0], "app"));
     auto callback_fn = Value::validated_to_function(ctx, args[1], "logger_callback");
-    app->sync_manager()->set_logger_factory(*new realm::node::SyncLoggerFactory(ctx, callback_fn));
+    
+    Protected<typename T::GlobalContext> protected_ctx(Context<T>::get_global_context(ctx));
+    Protected<FunctionType> protected_callback(ctx, callback_fn);
+
+    common::logger::Delegated show_logs = [=](int level, std::string message) { 
+        HANDLESCOPE(protected_ctx)
+
+        ValueType arguments[2] = {
+            Value::from_number(protected_ctx, level),
+            Value::from_string(protected_ctx, message)
+        };
+
+        Function::callback(protected_ctx, protected_callback, typename T::Object(), 2, arguments); 
+    };
+
+    auto sync_logger = common::logger::Logger::build_sync_logger(show_logs);
```

Comment:
This bit here seems to result in a memory leak.
`sync_logger` is a pointer.  The `SyncManager` stores this pointer, but doesn't take ownership over it.  And since `SyncClass<T>` is an N-API interface class, it is rather static, and can't really keep ownership of the pointer either.  Thus, no-one will deallocate it.
Can anyone with more insight into `SyncManager` shed some light on how to handle this issue (e.g., @kneth)?

Reply:
Good catch, @fronck. I made https://github.com/realm/realm-core/issues/4117 to track this.